pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiContributors.scala

```diff
@@ -0,0 +1,276 @@
+package weco.pipeline.transformer.tei.transformers
+
+import cats.instances.either._
+import cats.syntax.traverse._
+import weco.catalogue.internal_model.identifiers.IdState.{
+  Identifiable,
+  Unminted
+}
+import weco.catalogue.internal_model.identifiers.{
+  IdState,
+  IdentifierType,
+  SourceIdentifier
+}
+import weco.catalogue.internal_model.work.{
+  ContributionRole,
+  Contributor,
+  Person
+}
+import weco.pipeline.transformer.result.Result
+
+import scala.xml.{Elem, Node, Text, XML}
+
+object TeiContributors {
+
+  /**
+    * author nodes appear only within msItem. They contain the name of the Author of the work and optionally the id of the Author.
+    * The Id of the Author refers to two different authorities depending on whether the work is an Arabic manuscript or not
+    */
+  def authors(
+    node: Node,
+    isFihrist: Boolean
+  ): Result[List[Contributor[IdState.Unminted]]] =
+    (node \ "author")
+      .map { n =>
+        for {
+          authorInfo <- getLabelAndId(n)
+          (label, id) = authorInfo
+          res <- createContributor(
+            label = label,
+            id = id,
+            isFihrist = isFihrist,
+            ContributionRole("author"))
+        } yield res
+
+      }
+      .toList
+      .sequence
+
+  /**
+    * Scribes appear in the physical description section of the manuscript and can appear
+    * within a handNote tag with attribute "scribe" as in this example
+    * <physDesc>
+    *     <handDesc>
+    *         <handNote scope="minor" scribe="Scribe_A"> <locus target="#Wellcome_Batak_36801_1 #Wellcome_Batak_36801_2" >a 2-62, b 2-7; b 8-22</locus> Gorak-gorahan, etc. Southern form of ta.</handNote>
+    *         <handNote scope="minor" scribe="Scribe_B"> <locus target="#Wellcome_Batak_36801_3">b 23-35</locus>Another hand, which uses the northern ta.</handNote>
+    *         <handNote scope="minor" scribe="Scribe_C"> <locus target="#Wellcome_Batak_36801_9">b 44-45</locus>Another hand; careless writing, at first with northern ta, later southern ta.</handNote>
+    *         <handNote scope="minor" scribe="Scribe_D"> <locus target="#Wellcome_Batak_36801_10">b 45-56</locus>A different handwriting, using the southern form of ta but not the peculiar form of ya which is found in the text written by <persName>Datu Poduwon</persName>.</handNote>
+    *     </handDesc>
+    * </physDesc>
+    *
+    * or within a persName tag with role=scr inside the handNote tag
+    * <physDesc>
+    *    <handDesc>
+    *        <handNote scope="sole">
+    *            <persName role="scr">Mahādeva Pāṇḍe</persName>
+    *        </handNote>
+    *    </handDesc>
+    *</physDesc>
+    * If the scribe refers to a part or item of the manuscript, the handNote tag will have a locus node with the item or part ids that it refers to
+    *
+    */
+  def scribes(
+    xml: Elem,
+    workId: String): Result[Map[String, List[Contributor[Unminted]]]] =
```

Comment:
Can you explain what the keys of this map are? What are these string values?

Reply:
They are the work ids of the works they belong to. Will add in the comment